src/WSDataService.php

```diff
@@ -4,18 +4,52 @@
 
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
 
 /**
  * Service for processing WSData requests.
  */
 class WSDataService {
+  use StringTranslationTrait;
+
   protected $error;
+  protected $performance;
 
   /**
    * {@inheritdoc}
    */
   public function __construct(EntityTypeManagerInterface $entity_type_manager) {
     $this->entity_type_manager = $entity_type_manager;
+    $this->performance = [
+      'calls' => 0,
+      'runtime' => 0.0,
+      'log' => [],
```

Comment:
Generally do we log any of this to watchdog normally? Like as a debug level log entry?

Reply:
No, it enabled it logs, otherwise it's available to the API.  I'll add a method on the service to get this data.